nezha: some fixes to re-enable build and boot

threexc opened the pull request:
The commits currently present allow a build to complete, but when I copy the image to an SD card and try booting, I see:

```
U-Boot SPL 2024.01-rc1 (Oct 31 2023 - 08:53:02 +0000)
mxstatus=0xc0408000 mhcr=0x0000010b mcor=0x00000003 mhint=0x00004000
Trying to boot from MMC1
SPL size = 98304, sector = 192
Could not get FIT buffer of 1074176 bytes
        check CONFIG_SPL_SYS_MALLOC_SIZE
fdt_fixup_memory_banks: FDT_ERR_BADMAGIC
Failed to update DTB
resetting ...
reset not supported yet
### ERROR ### Please RESET the board ###
```

So more work needs to be done. Still, progress.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

threexc committed:
linux-allwinnerd1-dev: remove KERNEL_FEATURES_RISCV

Otherwise, we get error messages during build about config fragments not
being found because the kernel version isn't sufficiently new.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

threexc committed:
nezha-common: set KERNEL_IMAGETYPE to 'Image'

KERNEL_IMAGETYPE is derived from ARCH_DEFAULT_KERNELIMAGETYPE by
default, which is 'zImage' in oe-core for every architecture other than
x86. However, building for the nezha with this results in a "No rule to
make target 'zImage'" error for the kernel. Several of the other boards
in the meta-riscv layer set KERNEL_IMAGETYPE = "Image", so do the same
here.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

threexc committed:
nezha: increase SPL malloc to 2MB

Double the CONFIG_SPL_SYS_MALLOC_SIZE value to make room.

Without this, we see this error when booting:

|U-Boot SPL 2024.01-rc1 (Oct 31 2023 - 08:53:02 +0000)
|mxstatus=0xc0408000 mhcr=0x0000010b mcor=0x00000003 mhint=0x00004000
|Trying to boot from MMC1
|SPL size = 98304, sector = 192
|Could not get FIT buffer of 1074176 bytes
|        check CONFIG_SPL_SYS_MALLOC_SIZE
|fdt_fixup_memory_banks: FDT_ERR_BADMAGIC
|Failed to update DTB
|resetting ...
|reset not supported yet
|### ERROR ### Please RESET the board ###

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

kraj reviewed (commented):
Please add some color to why this is needed.

threexc commented:
> Please add some color to why this is needed.

Done, but now I've hit another error:

```
Hit any key to stop autoboot:  0                                                                                                                                                                                                             12:31:32 [32/1885]
switch to partitions #0, OK
mmc0 is current device
Scanning mmc 0:1...
Found U-Boot script /boot.scr.uimg
1399 bytes read in 1 ms (1.3 MiB/s)
## Executing script at 41900000
ethernet@4500000 Waiting for PHY auto negotiation to complete......... TIMEOUT !
ethernet@4500000 Waiting for PHY auto negotiation to complete......... TIMEOUT !
ethernet@4500000 Waiting for PHY auto negotiation to complete......... TIMEOUT !
322 bytes read in 2 ms (157.2 KiB/s)
Failed to load 'fitImage'
Wrong Image Format for bootm command
ERROR: can't get kernel image!
Failed to load 'fitImage'
ethernet@4500000 Waiting for PHY auto negotiation to complete......... TIMEOUT !
Failed to load 'uImage'
22950400 bytes read in 3791 ms (5.8 MiB/s)
Moving Image from 0x41000000 to 0x40200000, end=4183b000
ERROR: Did not find a cmdline Flattened Device Tree
Could not find a valid device tree
SCRIPT FAILED: continuing...
Card did not respond to voltage select! : -110
No EFI system partition
No EFI system partition
Failed to persist EFI variables
BootOrder not defined
EFI boot manager: Cannot load any image
```

I've got the network disconnected as I want to test booting from the SD card. I've also modified `recipes-bsp/u-boot/files/tftp-mmc-boot.txt` to include:

```
+# Finally, fallback to a MMC Image
+if load ${devtype} ${devnum}:${distro_bootpart} ${kernel_addr_r} Image; then
+  booti ${kernel_addr_r} - ${fdt_addr_r}
+fi;
```

but it seems weird that there'd be an error about a missing FDT - wouldn't the other boot steps referencing it also be wrong? Maybe I'm misunderstanding something.

threexc committed:
tftp-mmc-boot.txt: add Image + fdtcontroladdr boot step

For the Nezha, since we're now using the Image kernel format, rely on
the fdtcontroladdr holding the location of the dtb and add a boot option
in tftp-mmc-boot.txt for that case. This allows the board to boot after
falling through the other options. A longer-term solution should further
simplify this file since we know it won't be using a uImage and may not
need the other configs by default.

Signed-off-by: Trevor Gamblin <tgamblin@baylibre.com>

kraj reviewed (approved):
looks good
